tests/test_tasks.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2016 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+
+"""Module tests."""
+
+from __future__ import absolute_import, print_function
+
+import hashlib
+from os import listdir, path
+
+from invenio_sipstore.tasks import archive_sip
+
+
+def calculate_md5(filename):
+    """Calculate the MD5 of the content of a file."""
+    with open(filename) as fp:
+        return hashlib.md5(fp.read().encode('utf-8')).hexdigest()
+
+
+def test_task_bagit_file_generation(task_app, sip_with_file):
+    """Test sending a security message using Task module."""
+    archive_sip(sip_with_file.id)
+
+    dir = path.join('basepath', str(sip_with_file.id),
+                    sip_with_file.archived_at.isoformat())
+
+    root_content = listdir(dir)
+    assert 'bagit.txt' in root_content
+    assert 'bag-info.txt' in root_content
+    assert 'tagmanifest-md5.txt' in root_content
+    assert 'manifest-md5.txt' in root_content
+
+    assert len(listdir(path.join(dir, 'data', 'files'))) == 1
+
+
+def test_task_bagit_file_generation_bagit_txt(task_app, sip_with_file):
+    """Test of bagit.txt content."""
+    archive_sip(sip_with_file.id)
+
+    filename = path.join('basepath', str(sip_with_file.id),
```

Comment:
Once 'basepath' is fetched from config, you can create a fixture for it and cleanup afterwards, much like instance path:
https://github.com/inveniosoftware/invenio-sipstore/blob/master/tests/conftest.py#L49-L53
```python
@fixture()
def archive_path(app)
    tmp_path = tempfile.mkdtemp()
    path_orig = app.config['SIPSTORE_ARCHIVE_BASEPATH']
    app.config['SIPSTORE_ARCHIVE_BASEPATH'] = tmp_path
    yield path
    app.config['SIPSTORE_ARCHIVE_BASEPATH'] = path_orig
    shutil.rmtree(tmp_path)
```